### PR TITLE
ticket 10 done

### PR DESCRIPTION
**Pull request by yaahcd**

### CODE REVIEW

File: src/components/SearchBar.jsx [hunk 1]
```diff
@@ -1,64 +1,94 @@
...
+		getTopics()
+			.then((topics) => {
+				setTopicsList([{ slug: '' }, ...topics]);
+				setIsLoading(false);
+			})
+			.catch((err) => {
+				setError(err);
+				setIsLoading(false);
+			});
+	}, [topicsList]);
```
Comment (RichardNeat) [inline src/components/SearchBar.jsx:24]: Having this state in the dependency array will trigger the useEffect to run again... which sets the state again... which triggers the useEffect again etc! Pop a console.log into this component - are we getting any looping?

File: src/components/SearchBar.jsx [hunk 2]
```diff
@@ -1,64 +1,94 @@
...
-         <button onClick={handleClick} className="btn">Send</button>
-      </section>
-    )
-  }
+	if (isLoading) {
+		return <Loading />;
+	}
+
+	return (
+		<section className="searchBar">
```
Comment (RichardNeat) [inline src/components/SearchBar.jsx:62]: Great stuff on this input process, we are most of the way there to implementing this as a form. Consider how we can finish this off nicely to ensure it is fully controlled by react. The nc notes on controlled components may be helpful to refer back to here :)

Hint: Think of this as a form with an onSubmit 👍

**Review by RichardNeat — CHANGES_REQUESTED** [review]
Nice one! This is looking really nice so far. Just a few suggestions to polish this ticket in the comments below. Please take a look through and implement these changes.

Also regarding your user journey currently, it would be really nice if the URL would update with the topic page we are currently looking at (I realise this is to do with an earlier ticket, but I still think it would be really nice to address now). As a stretch goal it would also be really nice if the URL would update with the queries too. From a users perspective, this would be a great experience as they could create a sharable link to the exact view that they have on their screen.

Your SearchBar component is getting a little packed, you may want to consider if it could be abstracted into separate components. Remember to update your plan with this change if you do.

If you would like to discuss any of this please drop me a message, otherwise drop in another pr once you're happy with this task :)

**Review by NRMSMITH — COMMENTED** [review]
This is really lovely work Yara, well done.

Right now, your URL happens onChange, rather than onSubmit - I think that's the one thing I would change 😄 (send via DM to me to avoid being in the PR queue again!)

**Review by NRMSMITH — APPROVED** [review]
🚀 Merge! :)

**Pull request merged**